src/util/strencodings.cpp

```diff
@@ -30,8 +30,16 @@ static const std::string SAFE_CHARS[] =
 std::string SanitizeString(std::string_view str, int rule)
```

Comment:
what about changing the `SafeChars` enum to be an enum class instead? (enums auto-increment their value so you would just need to cast it to int here)